clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -269,7 +307,10 @@ private void checkDisconnects(long now) {
     private void failUnsentRequests() {
         // clear all unsent requests and fail their corresponding futures
         for (Map.Entry<Node, List<ClientRequest>> requestEntry: unsent.entrySet()) {
-            Iterator<ClientRequest> iterator = requestEntry.getValue().iterator();
+            List<ClientRequest> originalList = requestEntry.getValue();
```

Comment:
Nit: maybe just name them 'original' and 'copy'? Or better would probably be 'requests' and 'copyOfRequests'? You could probably avoid the need to name the copy if you change to a regular 'for' loop since we don't need the iterator anymore.